cmd/service.go

```diff
@@ -435,53 +435,66 @@ To keep the service running after the %s session completes, use %s.
 
 func runServiceStopCmd(cmd *cobra.Command, args []string) {
 	utils.LogTime("runServiceStopCmd stop")
+
+	stoppedChan := make(chan bool, 1)
+	var status db.StopStatus
+	var err error
+	var info *db.RunningDBInstanceInfo
+
+	spinner := display.StartSpinnerAfterDelay("", constants.SpinnerShowTimeout, stoppedChan)
+
 	defer func() {
 		utils.LogTime("runServiceStopCmd end")
 		if r := recover(); r != nil {
 			utils.ShowError(helpers.ToError(r))
 		}
 	}()
 
-	var status db.StopStatus
-	var err error
-
 	force := cmdconfig.Viper().GetBool(constants.ArgForce)
 	if force {
-		status, err = db.StopDB(force, db.InvokerService)
+		status, err = db.StopDB(force, db.InvokerService, spinner)
 	} else {
-		info, err := db.GetStatus()
+		info, err = db.GetStatus()
 		if err != nil {
+			display.StopSpinner(spinner)
```

Comment:
can we not do this in the defer - check if the spinner is nil and if not stop it

Reply:
The spinner has to be stopped before printing anything else.

If we do the stop in `defer` then we will have to float up the message as well